### PR TITLE
Add support for Football Match Status alerts

### PR DESCRIPTION
**Pull request by davidfurey**

### CODE REVIEW

File: common/src/main/scala/models/Notification.scala [hunk 1]
```diff
@@ -85,6 +86,121 @@ object ContentNotification {
...
+  debug: Boolean
+) extends Notification {
+  override def withTopics(topics: Set[Topic]): Notification = copy(topic = topics)
+}
+
+object FootballMatchStatusNotification {
+  import JsonUtils._
+
+  implicit val jf = new Format[FootballMatchStatusNotification] {
+    override def reads(json: JsValue): JsResult[FootballMatchStatusNotification] = {
```
Comment (alexduf) [inline common/src/main/scala/models/Notification.scala]: I suppose something is holding you back from using Play's macro but I can't spot it!

File: common/src/main/scala/models/Notification.scala [hunk 2]
```diff
@@ -85,6 +86,121 @@ object ContentNotification {
...
+        mapiUrl,
+        importance,
+        topic,
+        phase,
+        eventId,
+        debug
+      )
+    }
+
+    override def writes(o: FootballMatchStatusNotification): JsValue = JsObject(
```
Comment (alexduf) [inline common/src/main/scala/models/Notification.scala]: same here

File: common/src/main/scala/models/Notification.scala (hunk 1 above)
Comment (davidfurey) [inline common/src/main/scala/models/Notification.scala]: Greater than 22 fields in the case class (https://github.com/playframework/playframework/issues/3174)

**Comment by alexduf** [comment]
Hey @NathanielBennett I took over this branch/PR so I can't review it myself. Would you mind having a look at this one? thanks!

File: notification/app/notification/services/azure/GCMPushConverter.scala [hunk 3]
```diff
@@ -36,6 +36,7 @@ class GCMPushConverter(conf: Configuration) extends PushConverter {
     case bn: BreakingNewsNotification => toBreakingNews(bn)
     case el: ElectionNotification => toElectionAlert(el)
     case mi: LiveEventNotification => toLiveEventAlert(mi)
+    case ms: FootballMatchStatusNotification => toMatchStatusAlert(ms)
```
Comment (NathanielBennett) [inline notification/app/notification/services/azure/GCMPushConverter.scala]: You can't see it here but th the GoalAlert case on line 33 still needs removing

**Pull request merged**